ui/src/fixturemanager.cpp

```diff
@@ -494,6 +516,110 @@ void FixtureManager::updateRDMView()
+
+    if(m_doc->mode() == Doc::Design && m_highlightFixturesEnabled && m_parent->currentIndex() == 0)
+    {
+        // Turn on all selected fixtures
+        QSet<quint32>::const_iterator it;
+        for (it = newlySelectedFixtureIds.constBegin(); it != newlySelectedFixtureIds.constEnd(); ++it)
+        {
+            highlightFixture(*it);
+        }
+
+        // Turn all deselected fixtures off
+        for (it = deselectedFixtureIds.constBegin(); it != deselectedFixtureIds.constEnd(); ++it)
+        {
+            unHighlightFixture(*it);
+        }
+    }
+    else
+    {
+        // Turn all fixtures off
+        QHash<quint32, GenericDMXSource*>::iterator it;
+        foreach(const quint32 &fixtureId, m_fixtureToSourceMap.keys())
+        {
+            unHighlightFixture(fixtureId);
+        }
+    }
+
+    m_lastSelectedFixtureIds = selectedFixtureIds;
+}
+
+void FixtureManager::highlightFixture(quint32 id)
+{
+    Fixture* fxi = m_doc->fixture(id);
+    if (fxi == NULL)
+        return;
+
+    GenericDMXSource* source = new GenericDMXSource(m_doc);
+    m_fixtureToSourceMap.insert(id, source);
+
+    // TO-DO: check if this works for complex lights (Mac Quantum for instance)
+
+    // Set the color of every Fixture to white and intensity to 255
+    for (int i = 0; i < fxi->heads(); i++)
+    {
+        QLCFixtureHead head = fxi->head(i);
+
+        quint32 whiteChannel = head.channelNumber(QLCChannel::White, QLCChannel::MSB);
+        quint32 redChannel = head.channelNumber(QLCChannel::Red, QLCChannel::MSB);
+        quint32 greenChannel = head.channelNumber(QLCChannel::Green, QLCChannel::MSB);
+        quint32 blueChannel = head.channelNumber(QLCChannel::Blue, QLCChannel::MSB);
+
+        if(whiteChannel != QLCChannel::invalid())
+        {
+            source->set(fxi->id(), whiteChannel, 255);
+        }
+        else if(redChannel != QLCChannel::invalid() && greenChannel != QLCChannel::invalid() && blueChannel != QLCChannel::invalid())
+        {
+            source->set(fxi->id(), redChannel, 255);
+            source->set(fxi->id(), greenChannel, 255);
+            source->set(fxi->id(), blueChannel, 255);
+        }
+
+        quint32 panChannel = head.channelNumber(QLCChannel::Pan, QLCChannel::MSB);
+        if(panChannel != QLCChannel::invalid())
+        {
+            source->set(fxi->id(), panChannel, 127);
+        }
+
+        quint32 tiltChannel = head.channelNumber(QLCChannel::Tilt, QLCChannel::MSB);
+        if(tiltChannel != QLCChannel::invalid())
+        {
+            source->set(fxi->id(), tiltChannel, 127);
+        }
+
+        quint32 intensityChannel = head.channelNumber(QLCChannel::Intensity, QLCChannel::MSB);
+        source->set(fxi->id(), intensityChannel, 255);
+    }
+
+    source->setOutputEnabled(true);
+}
+
+void FixtureManager::unHighlightFixture(quint32 id)
+{
+    Fixture* fxi = m_doc->fixture(id);
+    if (fxi == NULL)
+        return;
+
+    if(!m_fixtureToSourceMap.contains(id))
```

Comment:
Code style: space after if/for/while/etc. Two rows above it is present...

Reply:
Fixed and noted for future reference 👌